ksql-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -98,6 +98,8 @@ ExecuteResult execute(
       KsqlConfig ksqlConfig,
       Map<String, Object> overriddenProperties);
 
+  List<PersistentQueryMetadata> getPersistentQueries();
```

Comment:
I will remove `numberOfPersistentQueries` in a follow-up PR since it touches quite a few files that otherwise wouldn't be touched. I promise it'll come right after :)

Reply:
ok

Reply:
Leaving this unresolved so that I can scan through and find all of my promised follow-ups

Reply:
#2570